### DISCUSSION TITLE
Use `-k` option to run `test_foo` but not `test_foo_bar`

### DISCUSSION THREAD

**Discussion by jwodder** [pytest-dev/pytest#10030] [Q&A] [answered]
I have a suite of test functions whose names include `test_foo`, `test_foo_bar`, `test_foo_quux`, `test_foo_the_other_way`, and so forth, and I want to tell pytest via the `-k` option (or any other applicable method) to only run `test_foo`.  Unfortunately, `-k test_foo` runs all of the mentioned tests, because their names start with "`test_foo`".  How can I get pytest to run just the one test?

**Comment by nicoddemus** [comment, marked as answer]
Unfortunately this is not possible currently, the only way to run just `test_foo` is to pass the full test node id, say `pytest tests/test_suite.py::test_foo`.

**Reply by jwodder** [reply]
And if all the tests I mentioned are in `tests/test_suite.py`, will that run just `test_foo` or all the `test_foo_stuff`s?

**Reply by nicoddemus** [reply]
Just `test_foo`.

When you give the full node id, pytest will run only that. 

`-k` on the other hand uses fuzzy matching, meaning if the test node id contains the text given to `-k`, it will be executed.

**Reply by oakkitten** [reply]
`foo and not foo_` will also work, although i prefer to just name my tests such that the name of one is never a part of the name of the other

**Reply by ramachandargadde** [reply]
try -k "test_foo-"

it will work, it will pick only test_foo

**Reply by twmr** [reply]
Better would be a commandline parameter (ore a new flag that changes the behavior of `-k`)  that allows the user to specify a nodeid regex. Then you could write `.*test_foo$` to just select the tests that end with `test_foo`.
